xdsl/dialects/utils/dimension_list.py

```diff
@@ -0,0 +1,68 @@
+from xdsl.dialects.builtin import I64, DenseArrayBase, IntegerType, i64
+from xdsl.irdl.declarative_assembly_format import (
+    AttributeVariable,
+    CustomDirective,
+    IRDLOperation,
+    ParsingState,
+    PrintingState,
+    irdl_custom_directive,
+)
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+from xdsl.utils.hints import isa
+
+
+def parse_empty_dimension_list(parser: Parser) -> bool:
+    if parser.parse_optional_characters("["):
+        parser.parse_characters("]")
+
+        return True
+
+    return False
+
+
+def print_dimension_list(printer: Printer, dims: DenseArrayBase[IntegerType]) -> None:
```

Comment:
Feels like we should have a `print_dimension_list` on the printer if we're adding it to the parser? I would recommend splitting this PR, first adding print/parse dimension list, and then adding the custom directive.